pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/source/PulsarSource.java

```diff
@@ -60,12 +61,15 @@ public void open(Map<String, Object> config) throws Exception {
         setupSerDe();
 
         // Setup pulsar consumer
-        this.inputConsumer = this.pulsarClient.newConsumer()
+        ConsumerBuilder<byte[]> consumerBuilder = this.pulsarClient.newConsumer()
                 .topics(new ArrayList<>(this.pulsarSourceConfig.getTopicSerdeClassNameMap().keySet()))
                 .subscriptionName(this.pulsarSourceConfig.getSubscriptionName())
-                .subscriptionType(this.pulsarSourceConfig.getSubscriptionType().get())
-                .ackTimeout(1, TimeUnit.MINUTES)
-                .subscribe();
+                .subscriptionType(this.pulsarSourceConfig.getSubscriptionType().get());
+
+        if (pulsarSourceConfig.getTimeoutMs() != null) {
```

Comment:
what if timeout is null

Reply:
Then the timeout is not set in the PulsarSource when the consumer is created.  Which is the behavior we want.